racket/collects/racket/private/list.rkt

```diff
@@ -409,17 +426,24 @@
                  (raise-argument-error 'name "procedure?" 0 f g))
                (unless (procedure? g)
                  (raise-argument-error 'name "procedure?" 1 f g))
-               (can-compose name 0 f f '())
-               (simple-compose f g)]
+               (cond
+                 [(id? f) g]
+                 [(id? g) f]
```

Comment:
I think these shortcuts have the same issue with not changing a function's arity in the case of `compose1`, and probably also the same naming issue.